stl/inc/stacktrace

```diff
@@ -300,19 +270,60 @@ public:
     }
 
 private:
-    static constexpr size_t _Max_frames = 0xFFFF;
-
     static constexpr bool _Noex_move = allocator_traits<_Alloc>::propagate_on_container_move_assignment::value
                                     || allocator_traits<_Alloc>::is_always_equal::value;
 
-    struct _Internal_t {
-        explicit _Internal_t() noexcept = default;
-    };
+    static constexpr bool _Noex_swap = allocator_traits<_Alloc>::propagate_on_container_swap::value
+                                    || allocator_traits<_Alloc>::is_always_equal::value;
+
+    static constexpr size_type _Max_frames = 0xFFFF;
+
+    // __declspec(noinline) to make the same behavior for debug and release.
+    // We force the current function to be always noinline and add its frame to skipped.
+
+    _NODISCARD __declspec(noinline) static basic_stacktrace
+        _Current(const unsigned long _Skip, const size_type _Max_depth, const allocator_type& _Al) noexcept {
+        // _Skip is adjusted outside (in `current`)
+        _TRY_BEGIN
+        basic_stacktrace _Result(_Al);
+
+        // <NOTICE> _Small_frame_limit may be subject to further decreasements to avoid stack burden, and
+        // <NOTICE> the cost introduced by the first try should be compensable from uninitialized resize.
+
+        constexpr size_type _Small_frame_limit = 128; // Sufficient in most cases.
+        stacktrace_entry _Buffer[_Small_frame_limit];
+        unsigned short _Caught =
+            __std_stacktrace_capture(_Skip, static_cast<unsigned long>((_STD min)(_Max_depth, _Small_frame_limit)),
+                reinterpret_cast<void**>(_Buffer), &_Result._Hash);
+        if (_Caught != _Small_frame_limit) { // actual size

```

Comment:
Should add `|| _Caught <= _Max_depth` to avoid spurious failure when `_Max_depth == _Small_frame_limit`; the same thing should be noticed in the contending version.